openpype/hosts/nuke/startup/custom_write_node.py

```diff
@@ -53,24 +58,79 @@
 }
 
 
-def main():
-    write_selected_nodes = [
-        s for s in nuke.selectedNodes() if s.Class() == "Write"]
+class WriteNodeKnobSettingPanel(nukescripts.PythonPanel):
+    """ Write Node's Knobs Settings Panel """
+    def __init__(self):
+        nukescripts.PythonPanel.__init__(self, "Set Knobs Value(Write Node)")
 
-    ext = None
-    knobs = knobs_setting["knobs"]
-    for knob in knobs:
-        if knob["name"] == "file_type":
-            ext = knob["value"]
-    for w in write_selected_nodes:
-        # data for mapping the path
-        data = {
-            "work": os.getenv("AVALON_WORKDIR"),
-            "subset": w["name"].value(),
-            "frame": "#" * frame_padding,
-            "ext": ext
-        }
-        file_path = temp_rendering_path_template.format(**data)
-        file_path = file_path.replace("\\", "/")
-        w["file"].setValue(file_path)
-        set_node_knobs_from_settings(w, knobs)
+        knobs_value = self.get_node_knobs_override()
+        # create knobs
+
+        self.typeKnob = nuke.Enumeration_Knob(
+            'override_subsets', 'override subsets', knobs_value)
+        # add knobs to panel
+        self.addKnob(self.typeKnob)
+
+    def process(self):
+        """ Process the panel values. """
+        write_selected_nodes = [
+            s for s in nuke.selectedNodes() if s.Class() == "Write"]
+
+        node_knobs = self.typeKnob.value()
+        ext = None
+        knobs = None
+        if node_knobs:
+            knobs = self.get_node_knobs_setting(node_knobs)
+            if not knobs:
+                nuke.message("No knobs value found in subset group..\nDefault setting will be used..")  # noqa
+                knobs = knobs_setting["knobs"]
+        else:
+            knobs = knobs_setting["knobs"]
+
+        for knob in knobs:
+            if knob["name"] == "file_type":
+                ext = knob["value"]
+        for w in write_selected_nodes:
+            # data for mapping the path
+            data = {
+                "work": os.getenv("AVALON_WORKDIR"),
+                "subset": w["name"].value(),
+                "frame": "#" * frame_padding,
+                "ext": ext
+            }
+            file_path = temp_rendering_path_template.format(**data)
+            file_path = file_path.replace("\\", "/")
+            w["file"].setValue(file_path)
+            set_node_knobs_from_settings(w, knobs)
+
+    def get_node_knobs_setting(self, value):
```

Comment:
I'd probably do a helper method that does :
```py
settings = [
            node
            for node in get_nuke_imageio_settings()["nodes"]["overrideNodes"]
        ]
        if not settings:
            return
```

or convert the two `get_node_knobs_setting` and `get_node_knobs_override` into one that returns a tuple of `(settings, overrides)`